src/charm.py

```diff
@@ -386,57 +426,23 @@ def _get_runner_manager(
             ),
         )
 
-    def _block_on_openstack_config(self, state: CharmState) -> bool:
-        """Set unit to blocked status on openstack configuration set.
-
-        Args:
-            state: The charm state instance.
-
-        Returns:
-            Whether openstack configuration is enabled.
-        """
-        if state.charm_config.openstack_clouds_yaml:
-            # Go into BlockedStatus as Openstack is not supported yet
-            self.unit.status = BlockedStatus(
-                "OpenStack integration is not supported yet. "
-                "Please remove the openstack-clouds-yaml config."
-            )
-            return True
-        return False
-
     @catch_charm_errors
     def _on_install(self, _: InstallEvent) -> None:
         """Handle the installation of charm."""
         state = self._setup_state()
 
-        if state.charm_config.openstack_clouds_yaml:
-            # Only build it in test mode since it may interfere with users systems.
-            if self.config.get(TEST_MODE_CONFIG_NAME) == "insecure":
-                self.unit.status = MaintenanceStatus("Building Openstack image")
-                github = GithubClient(token=state.charm_config.token)
-                image = openstack_manager.build_image(
-                    arch=state.arch,
-                    cloud_config=state.charm_config.openstack_clouds_yaml,
-                    github_client=github,
-                    path=state.charm_config.path,
-                    proxies=state.proxy_config,
-                )
-                instance_config = openstack_manager.create_instance_config(
-                    unit_name=self.unit.name,
-                    openstack_image=image,
-                    path=state.charm_config.path,
-                    github_client=github,
-                )
-                self.unit.status = MaintenanceStatus("Creating Openstack test instance")
-                instance = openstack_manager.create_instance(
-                    cloud_config=state.charm_config.openstack_clouds_yaml,
-                    instance_config=instance_config,
-                    proxies=state.proxy_config,
-                    dockerhub_mirror=state.charm_config.dockerhub_mirror,
-                    ssh_debug_connections=state.ssh_debug_connections,
-                )
-                logger.info("OpenStack instance: %s", instance)
-            self._block_on_openstack_config(state)
+        if state.instance_type == InstanceType.OPENSTACK:
```

Comment:
May I rename this to `cloud_type`? I think instance type could be confusing with flavor